src/main/java/org/opensearch/security/resources/ResourceAccessControlClient.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.resources;
+
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.security.spi.resources.FeatureConfigConstants;
+import org.opensearch.security.spi.resources.ResourceAccessActionGroups;
+import org.opensearch.security.spi.resources.client.ResourceSharingClient;
+import org.opensearch.security.spi.resources.sharing.ResourceSharing;
+import org.opensearch.security.spi.resources.sharing.ShareWith;
+import org.opensearch.security.spi.resources.sharing.SharedWithActionGroup;
+
+/**
+ * Access control client responsible for handling resource sharing operations such as verifying,
+ * sharing, revoking, and listing access to shareable resources.
+ *
+ * @opensearch.experimental
+ */
+public final class ResourceAccessControlClient implements ResourceSharingClient {
```

Comment:
We have significant test coverage holes in this code. I guess this is because the test coverage is supposed to come from the sample plugin - which however is not included in the coverage configuration. Is this correct? Can we change the coverage configuration to include this?

Reply:
Not sure what exactly is causing coverage drop but it also seems to be coming from files unrelated to this code. I will take a deeper look.

Reply:
looks like the coverage file did not process: https://app.codecov.io/gh/opensearch-project/security/commit/adf96f1d377ea673fbe5a31644fa731ffff85623

sample-plugin is already added as part of report-coverage task: https://github.com/DarshitChanpura/security/blob/3594c5a4213389f0a37dc6cc6a6f72d6f5e14fa7/.github/workflows/ci.yml#L118

Reply:
This commit https://github.com/opensearch-project/security/pull/5281/commits/6173cc77e1ea4d368f196d4ccffb5ae3fec9da5b should add coverage for files covered by sample-plugin.

Reply:
Coverage has now gone up significantly for patch. Feel free to close this thread if you do not have any more questions.